### PR TITLE
Zimperium mobile

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x]  Ready

missing ~~UT~~, ~~TPB~~, ~~outputs~~ and the ~~commands~~:
~~`zimperium-events-search`~~
~~`zimperium-devices-get-last-updated`~~
~~`zimperium-report-get`~~
`zimperium-app-upload-for-analysis` - will not be implemented for now.
As there are no examples in the docs, already contacted the vendor.

## Related Issues
fixes: https://github.com/demisto/etc/issues/23497

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by bakatzir** [comment]
@graphicool can you help with the logo https://www.zimperium.com/ ?

**Comment by graphicool** [comment]
@bakatzir , Ayal will slack you.

**Comment by bakatzir** [comment]
@Itay4 I've added a classifier, mapper, incidenttype and incidentfields as per Arseny's design in the reference issue.
Can you review them? as this is a first for me with 6.0

File: Packs/Zimperium/Classifiers/classifier-mapper-incoming-Zimperium.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+						"accessor": "",
+						"filters": [],
+						"root": "userPhoneNumber",
+						"transformers": []
+					},
+					"simple": ""
+				}
+			}
+		},
+		"dbot_classification_incident_type_all": {
```
Comment (Itay4) [inline Packs/Zimperium/Classifiers/classifier-mapper-incoming-Zimperium.json:87]: dont think this key is needed

File: Packs/Zimperium/Classifiers/classifier-Zimperium.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+	"id": "Zimperium",
+	"name": "Zimperium - Classifier",
+	"type": "classification",
+	"description": "Classifies Zimperium incidents.",
+	"defaultIncidentType": "",
```
Comment (Itay4) [inline Packs/Zimperium/Classifiers/classifier-Zimperium.json:6]: no default incident type?

File: Packs/Zimperium/Classifiers/classifier-mapper-incoming-Zimperium.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
+{
+	"id": "Zimperium - mapper",
+	"name": "Zimperium - Incoming Mapper",
+	"type": "mapping-incoming",
+	"version": -1,
+	"fromversion": "6.0.0",
+	"description": "Maps incoming Zimperium incident fields.",
+	"defaultIncidentType": "",
```
Comment (Itay4) [inline Packs/Zimperium/Classifiers/classifier-mapper-incoming-Zimperium.json:8]: no default incident type?

**Comment by Itay4** [comment]
make sure to set `defaultClassifier` and `defaultMapperIn` in the integration yml and add classifier `toversion: 5.9.9`

File: Packs/Zimperium/Classifiers/classifier-mapper-incoming-Zimperium.json (hunk 3 above)
Comment (bakatzir) [reply Packs/Zimperium/Classifiers/classifier-mapper-incoming-Zimperium.json:8]: Not AFAIK, they can choose to use the `Zimperium event` incidenttype, but not sure we want it as default. @Arsenikr WDYT?

File: Packs/Zimperium/Classifiers/classifier-Zimperium.json (hunk 2 above)
Comment (bakatzir) [reply Packs/Zimperium/Classifiers/classifier-Zimperium.json:6]: Not AFAIK, they can choose to use the Zimperium event incidenttype, but not sure we want it as default. @Arsenikr WDYT?

File: Packs/Zimperium/Integrations/Zimperium/Zimperium.py [hunk 4]
```diff
@@ -0,0 +1,661 @@
...
+            if event_id not in last_event_ids:  # check that event was not fetched in the last fetch
+                event_created_time = parse(event_data.get('persistedTime'))
+                incident = {
+                    'name': event_data.get('incidentSummary'),
+                    'occurred': event_created_time.strftime(timestamp_format),
+                    'severity': event_severity_to_dbot_score(event_data.get('severity')),
+                    'rawJSON': json.dumps(event_data)
+                }
+                incidents.extend([incident])
+                last_event_ids.extend([event_id])
```
Comment (yuvalbenshalom) [inline Packs/Zimperium/Integrations/Zimperium/Zimperium.py:573]: no order to allow you to filter by time?

File: Packs/Zimperium/Integrations/Zimperium/Zimperium_description.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+Zimperium uses the rsql to query for events, devices and users.
+Foe more information: https://github.com/jirutka/rsql-parser
```
Comment (yuvalbenshalom) [inline Packs/Zimperium/Integrations/Zimperium/Zimperium_description.md:2]:
```suggestion
For more information: https://github.com/jirutka/rsql-parser
```

File: Packs/Zimperium/Integrations/Zimperium/Zimperium_description.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+Zimperium uses the rsql to query for events, devices and users.
```
Comment (yuvalbenshalom) [inline Packs/Zimperium/Integrations/Zimperium/Zimperium_description.md:1]: this is the description to this product?

File: Packs/Zimperium/Integrations/Zimperium/Zimperium.py [hunk 7]
```diff
@@ -0,0 +1,661 @@
...
+        try:
+            shutil.copy(file_path, file_name)
+        except Exception:
+            raise Exception('Failed to prepare application for upload.')
+
+        try:
+            with open(file_path, 'rb') as file:
+                self._headers.update({'Content-Type': 'multipart/form-data'})
+                result = self._http_request(method='POST', url_suffix='/malware/public/upload/app',
+                                            headers=self._headers, files={'file1': file.read()}, timeout=240)
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Packs/Zimperium/Integrations/Zimperium/Zimperium.py:179]: This command is at risk, currently we get 500 from the server, I tried to increase the timeout but still failed.
see referenced issue for more details.

File: Packs/Zimperium/Integrations/Zimperium/Zimperium.py (hunk 4 above)
Comment (bakatzir) [reply Packs/Zimperium/Integrations/Zimperium/Zimperium.py:573]: You are right, I refactored accordingly.

File: Packs/Zimperium/Integrations/Zimperium/Zimperium_description.md (hunk 6 above)
Comment (bakatzir) [reply Packs/Zimperium/Integrations/Zimperium/Zimperium_description.md:1]: No, this is a detailed description if a customer wants to configure the fetch query or the queries for the search commands.

**Comment by bakatzir** [comment]
@yuvalbenshalom I'll do the docs after the yml review from alex. We are also currently still missing a few of them. I mailed the vendor I'll send one more reminder next week.

**Comment by bakatzir** [comment]
we added a default layout and pb as well:
Screen Shot 2020-07-02 at 11 29 25

**Comment by bakatzir** [comment]
rm app-upload due to 500 from zimperium
added `file` command as per @Arsenikr request.

**Comment by kirbles19** [comment]
@bakatzir Done.

**Pull request merged**